mani_skill/envs/tasks/control/hopper.py

```diff
@@ -0,0 +1,266 @@
+            ["hip", "knee", "waist"],
+            lower=-max_delta,
+            upper=max_delta,
+            damping=damping,
+            stiffness=stiffness,
+            force_limit=force_limit,
+            friction=friction,
+            use_delta=True,
+        )
+        pd_joint_delta_pos_ankle = PDJointPosControllerConfig(
+            ["ankle"],
+            lower=-max_delta / 3,
+            upper=max_delta / 3,
+            damping=damping,
+            stiffness=stiffness,
+            force_limit=force_limit,
+            friction=friction,
+            use_delta=True,
+        )
+        rest = PassiveControllerConfig(
+            [j.name for j in self.robot.active_joints if "root" in j.name],
+            damping=0,
+            friction=0,
+        )
+        return deepcopy_dict(
+            dict(
+                pd_joint_delta_pos=dict(
+                    body=pd_joint_delta_pos_body,
+                    ankle=pd_joint_delta_pos_ankle,
+                    rest=rest,
+                    balance_passive_force=False,
+                ),
+            )
+        )
+
+    def _load_articulation(self):
+        """
+        Load the robot articulation
+        """
+        loader = self.scene.create_mjcf_loader()
+        asset_path = str(self.mjcf_path)
+
+        loader.name = self.uid
+
+        # only need the robot
+        self.robot = loader.parse(asset_path)[0][0].build()
+        assert self.robot is not None, f"Fail to load URDF/MJCF from {asset_path}"
+        self.robot_link_ids = [link.name for link in self.robot.get_links()]
+
+    # planar agent has root joints in range [-inf, inf], not ideal in obs space
+    def get_proprioception(self):
+        return dict(
+            # don't include xslider qpos, for x trans invariance
+            qpos=self.robot.get_qpos()[:, 1:],
+            qvel=self.robot.get_qvel(),
+        )
+
+
+class HopperEnv(BaseEnv):
+    agent: Union[HopperRobot]
+
+    def __init__(self, *args, robot_uids=HopperRobot, **kwargs):
+        super().__init__(*args, robot_uids=robot_uids, **kwargs)
+
+    @property
+    def _default_sim_config(self):
+        return SimConfig(
+            gpu_memory_cfg=GPUMemoryConfig(
+                found_lost_pairs_capacity=2**25, max_rigid_patch_count=2**18
+            ),
+            scene_cfg=SceneConfig(
+                solver_position_iterations=15, solver_velocity_iterations=15
+            ),
+            # sim_freq=50,
+            control_freq=25,
+        )
+
+    @property
+    def _default_sensor_configs(self):
+        return self.agent._sensor_configs
+
+    @property
+    def _default_human_render_camera_configs(self):
+        return self.agent._sensor_configs
+
+    def _load_scene(self, options: dict):
+        loader = self.scene.create_mjcf_loader()
+        articulation_builders, actor_builders, sensor_configs = loader.parse(MJCF_FILE)
+        for a in actor_builders:
+            a.build(a.name)
+
+        # load in the ground
```

Comment:
for hopper (and also cartpole). I recommend we add a massive long white wall behind so that the rendered videos look a bit nicer with color 0.3, 0.3, 0.3 (this will match what ray-traced videos will look like since they include the ambient lighting  which is 0.3 0.3 0.3). Moreover with a big background wall if anyone wants to use depth image data they can and it won't be full of 0s (indicating no depth info since it goes to infinity).

Reply:
created new scene builder in mani_skill/utils/scene_builder/control/planar, for re-use for the other planar agents
new eval visual (512x512): 
https://github.com/user-attachments/assets/4a7cc102-52e0-4660-9356-cd0927878f33

Reply:
added wall to cartpole.py file directly, since it doesn't fit with planar scene builder or warrant its own scene builder

https://github.com/user-attachments/assets/8928292d-c58d-4b56-857c-126bdcf0ee5a

Reply:
ok we might want to play around with lighting in the future to make these look a bit brighter, but no need for now.